app/server/lib/DocApi.ts

```diff
@@ -1734,11 +1735,20 @@ export class DocWorkerApi {
     return docAuth.docId!;
   }
 
-  private _getDownloadOptions(req: Request, name: string): DownloadOptions {
+  private async _getDownloadFilename(req: Request, tableId?: string): Promise<string> {
+    // Query DB for doc metadata to get the doc data.
+    const doc = await this._dbManager.getDoc(req);
```

Comment:
If `title` is present in the query string, we don't need to query the DB. It's a small optimization, but could we refactor things so that this query only happens if `title` is unset?